API_OVERVIEW.md

```diff
@@ -0,0 +1,175 @@
+
+Version 2 of the Envoy API will evolve existing APIs and introduce new APIs to:
+
+* Allow for more advanced load balancing through load and resource utilization reporting to management servers.
+* Improve N^2 health check scalability issues by optionally offloading health checking to other Envoy instances.
+* Support Envoy deployment in edge, sidecar and middle proxy deployment models via changes to the listener model and CDS/SDS APIs.
+* Allow streaming updates from the management server on change, instead of polling APIs from Envoy. gRPC APIs will be supported
+  alongside JSON REST APIs to provide for this.
+* Ensure all Envoy runtime configuration is dynamically discoverable via API
+  calls, including listener configuration, certificates and runtime settings, which are today sourced from the filesystem. There
+  will still remain a static bootstrap configuration file that will specify items
+  unlikely to change during runtime, including the Envoy node identity, xDS
+  management server addresses, administration interface and tracing
+  configuration.
+* Revisit and where appropriate cleanup any v1 technical debt.
+
+## Status
+
+The LDS/CDS/EDS/RDS APIs are now frozen and will maintain backwards
+compatibility according to standard proto rules (e.g. new fields will not reuse
+tags, field types will not change, fields will not be renumbered, etc.).
+
+The remainder of the API (ADS, HDS, RLS, SDS, filter fragments other than HTTP
+connection manager, the bootstrap proto) are draft work-in-progress. Input is
+welcome via issue filing. Small, localized PRs are also welcome, but any major
+changes or suggestions should be coordinated in a tracking issue with the
+authors.
+
+Implementation work has begun and work items are tracked at
+[here](https://github.com/envoyproxy/envoy/issues?q=is%3Aopen+is%3Aissue+label%3A%22v2+API%22).
+
+New features that correspond to the v2 API are initially tracked in this
+repository. When they are agreed upon and the related PRs are merged, they
+should be closed out and a corresponding issue created in
+https://github.com/envoyproxy/envoy/ and tagged with `v2 API`. A reference to the
+closed issue should also be included.
+
+## Principles
+
+* [Proto3](https://developers.google.com/protocol-buffers/docs/proto3) will be
+  used to specify the canonical API. This will provide directly the gRPC API and
+  via gRPC-JSON transcoding the JSON REST API. A textual YAML input will be
+  supported for filesystem configuration files (e.g. the bootstrap file), in
+  addition to JSON, as a syntactic convenience. YAML file contents will be
+  internally converted to JSON and then follow the standard JSON-proto3
+  conversion during Envoy config ingestion.
+
+* xDS APIs should support eventual consistency. For example, if RDS references a
+  cluster that has not yet been supplied by CDS, it should be silently ignored
```

Comment:
may be rephrase to say traffic will be dropped (404s/503) until CDS update occurs..